core/node/state_keeper/src/keeper.rs

```diff
@@ -582,34 +593,51 @@ impl ZkSyncStateKeeper {
                     "L1 batch #{} should be sealed unconditionally as per sealing rules",
                     updates_manager.l1_batch.number
                 );
+
+                // Push the current block if it has not been done yet
+                if is_last_block_sealed {
+                    self.io
+                        .update_next_l2_block_timestamp(updates_manager.next_l2_block_timestamp());
```

Comment:
AFAIU, if `is_last_block_sealed`, we have next block params in `updates_manager`, so this will create a fictive block (= one w/o transactions), right? If so, I'd change the comment above and the log below to mention it explicitly. Also, it could be worth asserting at the callsite that after `process_l1_batch()` returns, `updates_manager` doesn't have next block params; this is necessary for correctness.